core/src/camera_rotation.cpp

```diff
@@ -0,0 +1,163 @@
+#include <knoting/camera_rotation.h>
+#include <knoting/components.h>
+#include <knoting/engine.h>
+#include <knoting/log.h>
+#include <knoting/scene.h>
+#include <knoting/spot_light.h>
+#include <fstream>
+#include <string_view>
+
+namespace knot {
+CameraRotation::CameraRotation(Engine& engine) : m_engine(engine) {}
+
+void CameraRotation::on_awake() {
+    m_engine.get_window_module().lock()->set_cursor_hide(m_lockState);
+
+    using namespace components;
+    auto sceneOpt = Scene::get_active_scene();
+    if (!sceneOpt) {
+        return;
+    }
+
+    Scene& scene = sceneOpt.value();
+    entt::registry& registry = scene.get_registry();
+    auto cameras = registry.view<Transform, EditorCamera, Name>();
+
+    for (auto& cam : cameras) {
+        auto goOpt = scene.get_game_object_from_handle(cam);
+        if (!goOpt) {
+            continue;
+        }
+
+        GameObject go = goOpt.value();
+        Transform& transform = go.get_component<Transform>();
+
+        auto rotation = transform.get_rotation_euler();
+
+        m_roll = rotation.x;
+        m_pitch = rotation.y;
+        m_yaw = rotation.z;
+    }
+}
+
+void CameraRotation::on_update(double m_delta_time) {
+    //= CHECK VALID INPUT ==
+    InputManager m_inputManager = m_engine.get_window_module().lock()->get_input_manager();
+
+    vec2d currentMousePos = m_inputManager.get_absolute_position();
+    if (glm::isinf(currentMousePos.x) || glm::isinf(currentMousePos.y)) {
```

Comment:
We don't need to use the `glm::` namespace specifier.

Reply:
isinf is a function in corectr_math.h and glm